torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3505,13 +3505,24 @@ std::tuple<Tensor, Tensor> householder_product_backward(const Tensor& grad, cons
   // If only single backward is run, we modify K in-place and exploit triangularity of the input.
   // With higher order derivatives we cannot rewrite the storage of K, hence we use much less efficient
   // out-of-place methods.
-  auto apply_householder_reflector = [&m](
-      int64_t k, const Tensor& v_full, const Tensor& t, Tensor& K, bool left = true, bool in_place = true) -> Tensor {
+  //
+  // if only first-order derivative is expected, we can modify K in-place for better performance
+  bool modify_K_in_place = !at::GradMode::is_enabled();
+
+  // TODO: replace this function with calls to LARFB
+  auto apply_householder_reflector = [&m, &modify_K_in_place](
```

Comment:
Nit: As `modify_K_in_place` is a bool it's even better to capture it by value. Same for `m` as it's an integer.

Reply:
Yeah, makes total sense, thanks!